### PR TITLE
Legend Properties

### PR DESCRIPTION
**Pull request by YuhanLu**
Add some rules to legend properties.
All properties tested in editor and all worked correctly.
Working on adding unit tests and determining some matching between vega and vega-lite.

### CODE REVIEW

File: test/compile/legend.test.ts [hunk 1]
```diff
@@ -36,6 +72,14 @@ describe('Legend', function() {
   });
 
   describe('properties.symbols', function() {
-    // FIXME test
+    it('should initialize if filled', function() {
+      const symbol = legend.properties.symbols({field: 'a'}, {}, parseUnitModel({
+          mark: "point",
```
Comment (YuhanLu) [inline test/compile/legend.test.ts]: Why this legend.properties has an error message of `Property 'properties' does not exist on type 'typeof "/home/zoe/Documents/vl2/vl/vega-lite/src/compile/legend"'.`?

Comment (kanitw) [inline test/compile/legend.test.ts]: add `export` keyword in front of `namespace` on line
https://github.com/vega/vega-lite/blob/zl/legend-properties/src/compile/legend.ts#L121

Comment (kanitw) [inline test/compile/legend.test.ts]: is it fixed?

Comment (YuhanLu) [inline test/compile/legend.test.ts]: Yes, I forgot to remove the line. Just fixed it.

File: site/docs/legend.md [hunk 2]
```diff
@@ -42,7 +42,21 @@ The `legend` property object supports the following properties:
...
 | format        | String        | The formatting pattern for axis labels. This is D3's [number format pattern](https://github.com/mbostock/d3/wiki/Formatting) for quantitative axis and D3's [time format pattern](https://github.com/mbostock/d3/wiki/Time-Formatting) for time axis. <span class="note-line">__Default value:__  derived from [`numberFormat`](config.html#format) config for quantitative axis and from [`timeFormat`](config.html#format) config for time axis.</span>. |
+| labelAlign    | String        | The alignment of the legend label. One of `"left"`, `"right"` or `"center"`. |
+| labelBaseline | String        | The position of the baseline of legend label. One of `"top"`, `"middle"` or `"bottom"`. |
+| labelColor    | String        | The color of the legend label. |
+| labelFont     | String        | The font of the label (e.g., `Helvetica Neue`). |
+| labelFontSize | Number        | The font size of the label, in pixels. The default value is 10. |
```
Comment (kanitw) [inline site/docs/legend.md]: inconsistent default value.

File: site/docs/legend.md [hunk 3]
```diff
@@ -39,10 +39,39 @@ If `legend` is `true`, default legend properties are applied.
 Legend properties can be overridden by setting `legend` to a legend property object.
 The `legend` property object supports the following properties:
 
+### General
+
 | Property      | Type          | Description    |
 | :------------ |:-------------:| :------------- |
 | orient        | String        | The orientation of the legend. One of `"left"` or `"right"`. This determines how the legend is positioned within the scene. <span class="note-line">__Default value:__  derived from [legend config](config.html#legend-config)'s `orient` (`"right"` by default).</span>|
-| title         | String        | The title for the legend. <span class="note-line">__Default value:__  derived from the field's name and transformation function applied e.g, "field_name", "SUM(field_name)", "BIN(field_name)", "YEAR(field_name)".</span> |
```
Comment (kanitw) [inline site/docs/legend.md]: @YuhanLu  why removing these 3 lines?

Comment (kanitw) [inline site/docs/legend.md]: Oh Just saw that you move them.  Good catch!

**Pull request merged**